### PR TITLE
move hyperstart code to libhyperstart

### PR DESCRIPTION
**Pull request by laijs**
Signed-off-by: Lai Jiangshan <jiangshanlai@gmail.com>

### CODE REVIEW

**Comment by laijs** [comment]
updated, ready for review

File: hypervisor/hypervisor.go [hunk 1]
```diff
@@ -31,13 +25,61 @@ func (ctx *VmContext) loop() {
...
+}
+
+func (ctx *VmContext) watchHyperstart(sendReadyEvent bool) {
+	timeout := time.AfterFunc(30*time.Second, func() {
+		ctx.Log(ERROR, "watch hyperstart timeout")
+		ctx.Hub <- &InitFailedEvent{Reason: "watch hyperstart timeout"}
+		ctx.hyperstart.Close()
+	})
+	for ctx.PauseState == PauseStateUnpaused {
+		_, err := ctx.hyperstart.APIVersion()
```
Comment (gao-feng) [inline hypervisor/hypervisor.go]: what happen if vm is paused after send out hyperstart message?

File: hyperstart/libhyperstart/serial.go [hunk 2]
```diff
@@ -0,0 +1,695 @@
...
+	stdoutPipe io.ReadCloser
+	stderrPipe io.ReadCloser
+}
+
+type serialBasedHyperstart struct {
+	sync.RWMutex
+	vmAPIVersion       uint32
+	closed             bool
+	lastStreamSeq      uint64
+	procs              map[pKey]*pState
```
Comment (gao-feng) [inline hyperstart/libhyperstart/serial.go]: Can't combine procs with ctx.vmExec? seems they are duplicate

Comment (laijs) [inline hyperstart/libhyperstart/serial.go]: ctx.vmExec is being removed later

**Comment by gao-feng** [comment]
LGTM

**Pull request merged**